pkg/csi/service/common/common_controller_helper.go

```diff
@@ -263,3 +264,21 @@ func IsOnlineExpansion(ctx context.Context, volumeID string, nodes []*cnsvsphere
 
 	return nil
 }
+
+// ExtractCsiControlOpFaultFromError extracts the fault type from the error of
+// CsiControlOp
+func ExtractCsiControlOpFaultFromError(ctx context.Context, err error) string {
+	log := logger.GetLogger(ctx)
+	log.Infof("ExtractCsiControlOpFaultFromError called with err %v", err)
+	if strings.Contains(err.Error(), prometheus.PrometheusFaultInvalidArgumentType) {
+		return prometheus.PrometheusFaultInvalidArgumentType
+	} else if strings.Contains(err.Error(), prometheus.PrometheusFaultNotFoundType) {
+		return prometheus.PrometheusFaultNotFoundType
+	} else if strings.Contains(err.Error(), prometheus.PrometheusFaultResourceInUseType) {
+		return prometheus.PrometheusFaultResourceInUseType
+	} else if strings.Contains(err.Error(), prometheus.PrometheusFaultManagedObjectNotFoundType) {
+		return prometheus.PrometheusFaultManagedObjectNotFoundType
+	} else {
+		return prometheus.PrometheusFaultOtherFaultType
+	}
```

Comment:
This looks like it should be an iteration over a loop, or better a lookup into a map